File: Spring/ParkingStatus/src/main/java/com/ParkingStatus/ParkingStatus/Service/StatusService.java
package com.ParkingStatus.ParkingStatus.Service;

import com.ParkingStatus.ParkingStatus.DataAccessService.Status.StatusDataAccessService;
import com.ParkingStatus.ParkingStatus.Models.Status.Status;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StatusService {

    @Autowired
    private final StatusDataAccessService statusDataAccessService;

    @Autowired
    public StatusService(StatusDataAccessService statusDataAccessService){
        this.statusDataAccessService = statusDataAccessService;
    }

    public List<Status> getAllStatuses(){
        return statusDataAccessService.selectAllStatuses();
    }

    public Status getStatusById(int id){
        return statusDataAccessService.selectStatusById(id);
    }

    public int createStatus(Status status){
       return statusDataAccessService.insertStatus(status);
    }

    public int updateStatus(int id, Status status){
       return statusDataAccessService.updateStatus(id,status);
    }

    public int deleteStatus(int id){
       return statusDataAccessService.removeStatus(id);
    }


}

File: Spring/ParkingStatus/src/js/parkingstatus.web/src/services/StatusService.ts
import axios from 'axios';

const STATUSES_REST_API_URL : string = 'http://localhost:8080/api/v1/status';

class StatusService{

    async getAllStatuses(){
       return axios.get(STATUSES_REST_API_URL);
    }

    async getStatusById(id: number){
        return axios.get(STATUSES_REST_API_URL + '/'+ id);
    }

    async createStatus(status: any){
        // console.log(status);

        return axios.post(STATUSES_REST_API_URL, status);
    }

    async updateStatus(id:number, status:any){
        return axios.post(STATUSES_REST_API_URL +'/'+ id, status);
    }

    async deleteStatus(id: number){
        return axios.delete(STATUSES_REST_API_URL + '/'+id);
    }

}

export default new StatusService();